src/main/scala/io/iohk/ethereum/consensus/ethash/difficulty/EthashDifficultyCalculator.scala

```diff
@@ -1,31 +1,46 @@
 package io.iohk.ethereum.consensus.ethash.difficulty
 
 import io.iohk.ethereum.consensus.difficulty.DifficultyCalculator
-import io.iohk.ethereum.domain.BlockHeader
+import io.iohk.ethereum.domain.{ BlockHeader, Blockchain }
 import io.iohk.ethereum.utils.BlockchainConfig
 
-class EthashDifficultyCalculator(blockchainConfig: BlockchainConfig) extends DifficultyCalculator {
-  import blockchainConfig.{difficultyBombContinueBlockNumber, difficultyBombPauseBlockNumber, homesteadBlockNumber, difficultyBombRemovalBlockNumber}
+class EthashDifficultyCalculator(blockchainConfig: BlockchainConfig, blockchain: Blockchain) extends DifficultyCalculator {
+  import blockchainConfig.{
+    difficultyBombContinueBlockNumber,
+    difficultyBombPauseBlockNumber,
+    homesteadBlockNumber,
+    difficultyBombRemovalBlockNumber,
+    byzantiumBlockNumber
+  }
 
   val DifficultyBoundDivision: Int = 2048
   val FrontierTimestampDiffLimit: Int = -99
   val ExpDifficultyPeriod: Int = 100000
   val MinimumDifficulty: BigInt = 131072
+  val RelaxDifficulty: BigInt = 3000000
 
   def calculateDifficulty(blockNumber: BigInt, blockTimestamp: Long, parentHeader: BlockHeader): BigInt = {
+    lazy val timestampDiff = blockTimestamp - parentHeader.unixTimestamp
 
     val x: BigInt = parentHeader.difficulty / DifficultyBoundDivision
     val c: BigInt =
       if (blockNumber < homesteadBlockNumber) {
         if (blockTimestamp < parentHeader.unixTimestamp + 13) 1 else -1
-      } else {
-        val timestampDiff = blockTimestamp - parentHeader.unixTimestamp
-        math.max(1 - timestampDiff / 10, FrontierTimestampDiffLimit)
+      } else if (blockNumber >= byzantiumBlockNumber) {
+        val ommersList = blockchain.getBlockBodyByHash(parentHeader.hash).map(_.uncleNodesList).getOrElse(Seq.empty)
```

Comment:
Why cant we just pass `parentBlock` (which have uncles) , instead `parentHeader` to this function (`calculateDifficulty`)?  
This way we wont introduce unnecessery dependancy on blockchain,  and also we wont be making that hidden side effect-y call in that function.

Reply:
that was my first approach but then were complications connected to it. this method is used in `validateDifficulty` in BlockHeaderValidatorSkeleton and in this class there is no blockchain passed/ in context nor whole parent. Only other option that I see is to create new method in `Ledger`, something similar to `getHeaderFromChainOrQueue ` but for body and pass it to all validators that are chained with `validateDifficulty` upwards, like: `validateBlockBeforeExecution` in `Ledger`, `validateBlockBeforeExecution` in `EthashValidators`, `validate`/ `validateDifficulty` in `BlockHeaderValidatorSkeleton`. wdyt?

Reply:
I've change `GetBlockHeaderByHash` to `GetBlockByHash`